tests/phpunit/tests/rest-api/wpRestTemplateAutosavesController.php

```diff
@@ -0,0 +1,425 @@
+		);
+
+		// Single.
+		$request  = new WP_REST_Request( 'OPTIONS', '/wp/v2/templates/' . self::TEST_THEME . '/' . self::TEMPLATE_NAME . '/autosaves/1' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount(
+			1,
+			$data['endpoints'],
+			'Failed to assert that the single autosave endpoints count is 1.'
+		);
+		$this->assertSame(
+			'view',
+			$data['endpoints'][0]['args']['context']['default'],
+			'Failed to assert that the default context for the single autosave endpoint is "view".'
+		);
+		$this->assertSame(
+			array( 'view', 'embed', 'edit' ),
+			$data['endpoints'][0]['args']['context']['enum'],
+			"Failed to assert that the enum values for the single autosave endpoint are 'view', 'embed', and 'edit'."
+		);
+	}
+
+	/**
+	 * @covers WP_REST_Template_Autosaves_Controller::get_items
+	 * @ticket 56922
+	 */
+	public function test_get_items() {
+		wp_set_current_user( self::$admin_id );
+		$autosave_post_id = wp_create_post_autosave(
+			array(
+				'post_content' => 'Autosave content.',
+				'post_ID'      => self::$template_post->ID,
+				'post_type'    => self::PARENT_POST_TYPE,
+			)
+		);
+
+		$request   = new WP_REST_Request(
+			'GET',
+			'/wp/v2/templates/' . self::TEST_THEME . '/' . self::TEMPLATE_NAME . '/autosaves'
+		);
+		$response  = rest_get_server()->dispatch( $request );
+		$autosaves = $response->get_data();
+
+		$this->assertCount(
+			1,
+			$autosaves,
+			'Failed asserting that the response data contains exactly 1 item.'
+		);
+
+		$this->assertSame(
+			$autosave_post_id,
+			$autosaves[0]['wp_id'],
+			'Failed asserting that the ID of the autosave matches the expected autosave post ID.'
+		);
+		$this->assertSame(
+			self::$template_post->ID,
+			$autosaves[0]['parent'],
+			'Failed asserting that the parent ID of the autosave matches the template post ID.'
+		);
+		$this->assertSame(
+			'Autosave content.',
+			$autosaves[0]['content']['raw'],
+			'Failed asserting that the content of the autosave is "Autosave content.".'
+		);
+	}
+
+	/**
+	 * @covers WP_REST_Template_Autosaves_Controller::get_item
+	 * @ticket 56922
+	 */
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+
+		$autosave_post_id = wp_create_post_autosave(
+			array(
+				'post_content' => 'Autosave content.',
+				'post_ID'      => self::$template_post->ID,
+				'post_type'    => self::PARENT_POST_TYPE,
+			)
+		);
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/templates/' . self::TEST_THEME . '/' . self::TEMPLATE_NAME . '/autosaves/' . $autosave_post_id );
+		$response = rest_get_server()->dispatch( $request );
+		$autosave = $response->get_data();
+
+		$this->assertIsArray( $autosave, 'Failed asserting that the autosave is an array.' );
+		$this->assertSame(
+			$autosave_post_id,
+			$autosave['wp_id'],
+			sprintf(
+				'Failed asserting that the autosave id is the same as %s.',
+				$autosave_post_id
+			)
```

Comment:
No need for `sprintf()` here, we can just interpolate:


```suggestion
			"Failed asserting that the autosave id is the same as $autosave_post_id."
```
Applies elsewhere in the PR (except for the static property / `::class` references).

Reply:
Fixed.